app/src/main/java/xyz/malkki/neostumbler/utils/tileJson.kt

```diff
@@ -0,0 +1,36 @@
+package xyz.malkki.neostumbler.utils
+
+import android.util.Log
+import java.io.IOException
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.json.JSONObject
+
+fun getTileJsonLayerIds(tileJson: String, callback: (List<String>) -> Unit) {
+    val layerIds = mutableListOf<String>()
+
+    OkHttpClient().newCall(Request.Builder().url(tileJson).build()).enqueue(object : Callback {
```

Comment:
An existing OkHttpClient instance should be reused here so that we can benefit from caching and connection pooling. In `MapViewModel` it's available from `httpClient.value`.

Also coroutines could be used here

Reply:
I reused it.